Demo/Classes/MBProgressHUD+GlobalHUD.m

```diff
@@ -0,0 +1,103 @@
+//
+//  MBProgressHUD+GlobalHUD.m
+//  Remind101
+//
+//  Created by Rex Fenley on 11/9/13.
+//
+
+// This code is distributed under the terms and conditions of the MIT license.
+
+// Copyright (c) 2013 Remind101, Rex Fenley
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MBProgressHUD+GlobalHUD.h"
+
+#import <objc/runtime.h>
+
+@implementation MBProgressHUD (GlobalHUD)
+
+static void * const kHUDCountStorageKey = (void *)&kHUDCountStorageKey;
+static MBProgressHUD * globalHUD;
+
++ (void)renewGlobalHUD
+{
+    NSArray *windows = [[UIApplication sharedApplication] windows];
+    UIWindow *window = [windows lastObject];
```

Comment:
@rex-remind101 thanks for the quick reply! I'm happy to clarify :)

In a project I'm working on I've also decided to use something similar to global HUD instead of supplying it with a view. Here is a short version of what I have:

```swift
extension MBProgressHUD {

  static func show() -> MBProgressHUD {
    if let hud = MBProgressHUD(forView: firstWindow) {
      return hud
    }
    return showHUDAddedTo(firstWindow, animated: true)
  }

  // MARK: Internal

  private static var firstWindow: UIWindow {
    return UIApplication.sharedApplication().windows.first!
  }

}
```

As you can see I'm adding HUD to `UIApplication.sharedApplication().windows.first` and I found this enough. In your case I see you are trying to use `UIApplication.sharedApplication().windows.last` and if it can't be used you are taking iterating from the end of the array to find a window to add the HUD to.

The question is in which situations my code will work incorrectly :)